File: src/Server/services/menu.api/Menu.API/DataTransferObjects/FoodDto.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Menu.API.DataTransferObjects
{
    public class FoodDto
    {
        [JsonProperty("id")]
        public Guid Id { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("pictures")]
        public List<FoodPictureDto> Pictures { get; set; }

        [JsonProperty("price")]
        public decimal Price { get; set; }

        [JsonProperty("category")]
        public CategoryDto CategoryDto { get; set; }

        [JsonProperty("categoryId")]
        public Guid CategoryId { get; set; }

        [JsonProperty("currency")]
        public string Currency { get; set; }

        public FoodDto Clone()
        {
            return (FoodDto)this.MemberwiseClone();
        }
    }
}

File: src/backend/web/web.client/web-app/src/api/dtos/FoodDto.ts
export interface IFoodDto {
    id:              string;
    name:            string;
    description:     string;
    pictures:        IPicture[];
    price:           number;
    category:        ICategory;
    categoryId:      string;
    currency:        string;
}

export interface ICategory {
    id:    string;
    name:  string;
    color: string;
}

export interface IPicture {
    id:       string;
    filePath: string;
}